rust/src/lib/nm/profile.rs

```diff
@@ -148,6 +148,33 @@ fn can_skip_activation(
     nm_conn: &NmConnection,
     exist_nm_conns: &[NmConnection],
 ) -> bool {
+    // if the controller is desired to be down or absent, activating the
+    // connection on the port will risk making the controller activate again,
+    // therefore skip the activation on the port
+    log::info!("merged_iface.current: {}, merged_iface.for_apply: {}, merged_iface.merged.is_up(): {}", merged_iface.current.is_none(), merged_iface.for_apply.is_some(), merged_iface.merged.is_up());
+    if let Some(desired_iface) = merged_iface.for_apply.as_ref() {
+        if let (Some(ctrl_iface), Some(ctrl_type)) = (
+            desired_iface.base_iface().controller.as_deref(),
+            desired_iface.base_iface().controller_type.as_ref(),
+        ) {
+            if let Some(merged_ctrl_iface) =
+                merged_ifaces.get_iface(ctrl_iface, ctrl_type.clone())
+            {
+                if merged_ctrl_iface.for_apply.is_some()
+                    && (merged_ctrl_iface.merged.is_absent()
+                        || merged_ctrl_iface.merged.is_down())
```

Comment:
Not sure if we should call `merged_ctrl_iface.desired.as_ref().unwrap.is_absent()` or `merged_ctrl_iface.desired.as_ref().unwrap.is_down()` here ?

Reply:
unwrap() is never allowed in library code. Current code is OK as you already checked `for_apply()`, then the `merged` will hold `desired` state.